### PR TITLE
Added lenientValidation support to REMReM generate

### PR DESCRIPTION
**Pull request by raja-maragani**
### Applicable Issues

https://github.com/eiffel-community/eiffel-remrem-generate/issues/171

### Description of the Change

Added Lenient Validation support to REMReM Generate, The Lenient Validation will perform the only mandatory field validation and non-mandatory validation failures will place in Eiffel message as a new property(remremGenerateFailures)

### Benefits

The REMReM generate validation will not fail on non-fetal Eiffel message field issues. This feature is an option to the user, the user can decide the enable/disable the Lenient Validation. By default it is disabled.

### Sign-off

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

Signed-off-by: @raja-maragani

### CODE REVIEW

**Comment by magnusbaeck** [comment]
This PR depends on https://github.com/eiffel-community/eiffel-remrem-semantics/pull/128 where I just posted some comments. I'll hold off the review until that PR progresses.

**Comment by raja-maragani** [comment]
> even though we declare the default values in the API, it is always prefered to be sent by the users.
> Can we add that part in documentation.
> 
> Also the CLI is non-backwards compatible if the lv is not provided. so should that be also handled ?

CLI works as backward compatibility.  API default values from the user is a separate part issue 
https://github.com/eiffel-community/eiffel-remrem-generate/issues/161

File: wiki/markdown/usage/generate-cli.md [hunk 1]
```diff
@@ -20,6 +20,8 @@ usage: java -jar
 
  -mp,--messaging_protocol    name of messaging protocol to be used, e.g. eiffelsemantics
 
+ -lv,--lenientValidation     lenientValidation will perform the only mandatory field validation and non-mandatory validation failures will place in Eiffel message as a new property(remremGenerateFailures)
```
Comment (Umadevi-Kapu) [inline wiki/markdown/usage/generate-cli.md:23]: only on

Comment (Umadevi-Kapu) [inline wiki/markdown/usage/generate-cli.md:23]: in customData

File: wiki/markdown/usage/lenientvalidation.md [hunk 2]
```diff
@@ -0,0 +1,199 @@
+## Lenient Validation
+The Lenient validation introduced in REMReM Generate since the version 2.1.0
+
+Using the lenient validation user continue the Eiffel generate with non-fatal fields in the message.
```
Comment (Umadevi-Kapu) [inline wiki/markdown/usage/lenientvalidation.md:4]: eiffel event generation

File: wiki/markdown/usage/lenientvalidation.md [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+This validation flexibility applies to only Eiffel's message optional fields. see
+ [Eiffel REMReM semantics](https://github.com/eiffel-community/eiffel/tree/master/schemas)
+for more about mandatory and optional fields.
+
+The Lenient validation will perform the only mandatory field validation and non-mandatory validation failures will place in Eiffel message as a new property(remremGenerateFailures)
```
Comment (Umadevi-Kapu) [inline wiki/markdown/usage/lenientvalidation.md:9]: customData

File: wiki/markdown/usage/lenientvalidation.md [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+         {
+            "type":"ENVIRONMENT",
+            "target":"aaaaaaaa-bbbb-5ccc-8ddd-eeeeeeeeeee3"
+         }
+      ]
+   }
+]
+```
+
+The lenient validation created new message property with "remremGenerateFailures"
```
Comment (Umadevi-Kapu) [inline wiki/markdown/usage/lenientvalidation.md:177]: new element in customData

**Comment by raja-maragani** [comment]
If no other comments, I will merge the PR.

**Pull request merged**